bgpd/bgp_io.c

```diff
@@ -235,6 +236,7 @@ static int bgp_process_reads(struct thread *thread)
 			stream_set_endp(pkt, pktsize);
 
 			frr_with_mutex(&peer->io_mtx) {
+				frrtrace(2, frr_bgp, packet_read, peer, pkt);
```

Comment:
This trace call is inside a block that's holding a mutex - but that's not necessary, is it? the data that would be captured doesn't need the mutex's protection. I'm asking because these initial examples will be copied in other daemons, and it'd be good to establish the patterns that you think will work best.

Reply:
No, it's not needed. I chose to put the tracepoint as close as possible to the action that it was reporting on, but you're right, it doesn't need to be inside the mutex. I'll move it.